Add methods to get and set bits of collision layers and masks for TileMaps (3.0)

MattUV opened the pull request:
Add four methods to the TileMap node to make collision layers and masks be modified bit by bit (like PhysicBody2Ds and RayCast2Ds:

 * set_collision_layer_bit()
 * set_collision_mask_bit()

 * get_collision_layer_bit()
 * get_collision_mask_bit()

To comply with collision layers' renaming from godotengine#5696 , the method names are NOT set/get_layer_mask_bit()

MattUV committed:
Add set/get_collision_layer/mask_bit() to TileMaps

Add four methods to the TileMap node to make collision layers and masks be modified bit by bit (like PhysicBody2Ds and RayCast2Ds:

 * set_collision_layer_bit()
 * set_collision_mask_bit()

 * get_collision_layer_bit()
 * get_collision_mask_bit()

To comply with collision layers' renaming from https://github.com/godotengine/godot/issues/5696 , the method names are NOT set/get_layer_mask_bit()

MattUV commented:
I really wonder if we should keep two methods that do exactly the same thing, or if we should just remove one of them.
If we keep the two methods, I will correct the code as in the 2.1 pull request.

bojidar-bg commented:
Not sure, I think one of set/get_collision_layer or set/get_layer_mask went away in 3.0... Refs: 0ad9939603a3d659f3f9cfba095a2e0c082483e8

eon-s commented:
This was mentioned here https://github.com/godotengine/godot/issues/5696#issuecomment-248070571, maybe @reduz can say if keep or get rid of the old methods now that you found a way to make it consistent with bodies/rays.

MattUV commented:
From what I see in the discussion, and with the comments, set_collision_layer and set_collision_mask has been chosen. I will change the code based on that, and modify the commit when it's done.

MattUV commented:
To summarize: I have just added the methods with "bit" at the end.
As the TileMap code follows what @reduz did in 0ad9939 , I did NOT add:
* set_layer_mask()
* get_layer_mask()
* set_layer_mask_bit()
* get_layer_mask_bit()

Hence, I modified the pull and commit's name and description.

Note that I did NOT modify the property `collision_layers` into `collision_layer` (line 1288) as it was made in 0ad9939, because I could not find the file `prop_rename.txt` which is on @kbake 's repo.

akien-mga commented:
> Note that I did NOT modify the property collision_layers into collision_layer (line 1288) as it was made in 0ad9939, because I could not find the file prop_rename.txt which is on @kbake 's repo.

That `prop_rename.txt` file is not important, it was just a helper where @reduz wrote the renamings he was doing to help with later making the 2to3 conversion tool.
